backend/msgpack/writer.go

```diff
@@ -91,6 +96,10 @@ func (w *writer) Write(
 	mu unaggregated.MetricUnion,
 	pl policy.PoliciesList,
 ) error {
+	if w.rand.Float64() < w.samplingRate {
```

Comment:
Some benchmarks I ran show that a call to `Float64` takes on the order of 5ns when using a lock `Rand`: https://github.com/jeromefroe/golang_benchmarks#rand